napari/conftest.py

```diff
@@ -549,3 +551,11 @@ def pytest_collection_modifyitems(session, config, items):
                 index = i
         test_order[index].append(item)
     items[:] = list(chain(*test_order))
+
+
+@pytest.fixture(autouse=True)
+def disable_notif_thread(monkeypatch):
```

Comment:
Not sure what our standard has been until now, but I think it's really important to add comments/docstrings to explain skips or patches on tests because they change the behavior of the things you're trying to test. Without explanatory comments, it quickly becomes hard to understand why the changes exist and if/when it's safe to remove them.

Therefore, it would be good to add a comment here about what this does (disables hiding the notification dialog) and why we don't want to test it (because it introduces timing sensitivity that may cause problems at teardown).

Reply:
> because it introduces timing sensitivity that may cause problems at teardown

it is not a problem on teardown, but long after teardown. It is 5 seconds of waiting. I try to explain it in the comment. 



> If I understand this change correctly, it will apply to every test.

Yes. Because notification could be emitted in another test. But because of its nature test could be segfaulted before printing a real error message. 

Could you check if the current docstring properly explains the problem?